src/assets/CollectibleDetectionController.ts

```diff
@@ -381,12 +390,10 @@ export class CollectibleDetectionController extends BaseController<
                 collectionImage: collection.image_url,
               },
             );
-            await this.addCollectible(
-              address,
-              token_id,
-              collectibleMetadata,
-              true,
-            );
+            await this.addCollectible(address, token_id, collectibleMetadata, {
+              userAddress: selectedAddress,
+              chainId: chainId as string,
```

Comment:
Nit: We should not need to cast here. If it's possible for this to be something other than a string, we need to deal with that here. If it's not, then the type is wrong and we should fix that instead.

Reply:
The chain ID normalization above is responsible for this. It is guaranteed to be a string here but the normalization is done in a way TypeScript's inference is having trouble with. I've just prototyped a solution that I can prepare as a follow up PR later.

Reply:
Basically it's to move the normalization to a utility function, and adjust the conditions to help TypeScript's inference along a bit better, and add a "throw" case for invalid chain IDs.

Reply:
I see, this type and normalization is also present in the `GasFeeController`, so we can use that util there.